File: code-coaches/medium/day-of-the-week/day_of_the_week.rb
require 'date'

def get_day_of_week(date_string)
  begin
    date = Date.strptime(date_string, '%m/%d/%Y')
  rescue ArgumentError
    date = Date.strptime(date_string, '%B %d, %Y')
  end
  date.strftime('%A')
end

def main
  date_string = gets.chomp
  puts get_day_of_week(date_string)
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/day-of-the-week/day_of_the_week.py
from datetime import datetime


def get_day_of_week(date_string):
    try:
        date = datetime.strptime(date_string, "%m/%d/%Y")
    except ValueError:
        date = datetime.strptime(date_string, "%B %d, %Y")
    return date.strftime("%A")


def main():
    date_string = input()
    day_of_week = get_day_of_week(date_string)
    print(day_of_week)


if __name__ == "__main__":
    main()
